winpr/libwinpr/clipboard/dirent.c

```diff
@@ -0,0 +1,91 @@
+
+#include <stdio.h>
+#include "dirent.h"
+
+#include "../log.h"
+#define TAG WINPR_TAG("clipboard.synthetic.file")
+
+DIR* opendir(const char* name)
+{
+	DIR* dir = NULL;
+	WIN32_FIND_DATAW FindData;
+	char namebuf[MAX_PATH];
+	WCHAR* file = NULL;
+	int nLen = 0;
+	int nRet = 0;
+
+	sprintf(namebuf, "%s\\*.*", name);
+	nLen = strlen(namebuf);
+
+	do
```

Comment:
Hint: You can just do ` { code }` instead of `do { code } while(0);` for scoping

Reply:
Here's a trick to make sure you don't forget to free memory.